### DISCUSSION TITLE
why symmetric contraction doesn't work for some hidden irreps?

### DISCUSSION THREAD

**Discussion by Fadelis98** [ACEsuit/mace#42] [Q&A] [answered]
Although already mentioned in #36 that all degrees should have the same number of channels, I find it doesn't work for some certain irreps like 0o,1e. The error occurs in `cg.py`: line `117 current_ir = wigners[0][0]`, list out of range. Or in line `132 out += [last_ir, stack]`, last_ir not defined.

In the README guidance, it seems that these irreps are avoided on purpose, but I didn't see the reason in the MACE paper, could anyone  give me some insight?

**Comment by ilyes319** [comment, marked as answer]
Hey,

MACE currently uses a restricted part of the O(3) that maximises both efficiency and accuracy.  This group corresponds to irreps of alternating parity, thus why we are not creating these irreps. In the current form of the generalized cg generation, we avoid these paths on purpose. A new algo for generating the cg will be soon available where one will be able to add these.

If for some reason you want these irreps in your final output, you can always take tensorproducts after the readout blocks to create them.
